test/mainController.test.ts

```diff
@@ -154,6 +114,23 @@ suite('MainController Tests', async () => {
         }
     });
 
+    // Renamed file event test
+    test('onDidCloseTextDocument should call renamedDoc function when rename occurs' , done => {
```

Comment:
done is unnecessary here since this is a sync method. Just remove the try/catch and if an error is thrown it'll fail the test.

Reply:
The way the tests have been written, the done is required.

Reply:
Why? I don't see how this wouldn't accomplish the same thing since you're never awaiting anything and don't have any callbacks.

```
// A renamed doc constitutes an openDoc event directly followed by a closeDoc event
mainController.onDidOpenTextDocument(newDocument);
mainController.onDidCloseTextDocument(document);

// Verify renameDoc function was called
connectionManager.verify(x => x.transferFileConnection(TypeMoq.It.isAny(), TypeMoq.It.isAny()), TypeMoq.Times.once());
assert.equal(docUriCallback, document.uri.toString());
assert.equal(newDocUriCallback, newDocument.uri.toString());
```

Reply:
I don't want to change old tests, but when I tried removing the `done`, the tests timed out. The mocha interface is also very flaky (especially the TDD interface we use, and there are tons of bugs on the repo), so I don't want to change them. If you can figure out how to not time out the tests while using the `done`, feel free to make the change.

Reply:
Well, there's no reason to have it be async so I'd just remove that and what I had above should be fine. But up to you if you don't want to try and fix it.